ocaml/testsuite/tests/typing-layouts-float64/parsing.ml

```diff
@@ -47,7 +47,7 @@ Error: This type float# should be an instance of type ('a : value)
        The layout of float# is float64, because
          it equals the primitive value type float#.
        But the layout of float# must be a sublayout of value, because
-         a type argument defaults to layout value.
+         a type argument of list defaults to layout value.
```

Comment:
Actually, on further thought, something is off here.

`list` is imported. We don't really care, here, whether its argument's layout was defaulted or annotated. I realize this is an orthogonal change to the one you're aiming for here, but I think I'd rather this line read

```
the type argument of imported type list has layout value
```

That is, we just state that it has layout value. The "imported" part is meant to convey that this fact isn't up for debate or inference. But maybe that part is less clear.

On the other hand, the change you made would be great for

```
type 'a t
val f : 'a -> 'a t
let errors = f (Float_u.of_float 3.14)
```

where the fact that defaulting happened is relevant -- precisely because the type is local.

How to achieve this? Not quite sure. You might want a `Imported_type_argument` constructor for one of the reason types (it would apply to any layout, not just values). I know that the layouts for the arguments of `list`, `option`, and `array` all are hard-coded in `predef.ml`. You could use `Imported_type_argument` there. Harder would be sorting this out for proper imported types. The current treatment (in subst.ml) that changes all the reasons to `Imported` upon writing to a .cmi file at the end of type-checking would be hard to rewrite in this way. Perhaps better would be to do it upon *loading* a cmi file... which might just work by adding some logic to `Env.store_type` to update the reasons there. (That would also likely update the reasons for predef types, so you might be able to get away with putting dummy reasons in predef. Worth experimenting.)